projects/flimnap/main.cc

```diff
@@ -297,37 +324,58 @@ int main(void) {
         float steer_torque = 0.0f;
         (void)kistler_torque;
 #else // defined(FLIMNAP_ZERO_INPUT)
-        const float inertia_torque = -handlebar_model.torque(model_t::get_state_part(bicycle.full_state()));
+        const float inertia_torque = -handlebar_inertia.torque(model_t::get_state_part(bicycle.full_state()));
         float steer_torque = kistler_torque - inertia_torque;
 #endif // defined(FLIMNAP_ZERO_INPUT)
 
         // simulate bicycle
         bicycle.set_v(v);
+        { // perform variant specific code for creating input/measurement
+#if defined(USE_BICYCLE_AREND_MODEL)
+            // BicyleArend uses a different output/measurement vector definition
+            const float steer_rate = util::encoder_rate(encoder_steer);
+
+            model_t::input_t u = model_t::input_t::Zero();
+            model_t::set_input_element(u, model_t::input_index_t::roll_torque, roll_torque);
+            model_t::set_input_element(u, model_t::input_index_t::steer_torque, steer_torque);
+
+            model_t::measurement_t z = model_t::measurement_t::Zero();
+            bicycle.model().set_output_element(z, bicycle.model().output_index_t::steer_angle, steer_angle);
+            bicycle.model().set_output_element(z, bicycle.model().output_index_t::steer_rate, steer_rate);
+
+            bicycle.update_dynamics(u, z);
+#else
 #if !defined(USE_BICYCLE_KINEMATIC_MODEL)
-        if (bicycle.v() < assistance_velocity_limit) {
-            const float interp = bicycle.v() / assistance_velocity_limit;
-            const model_t::input_t u = controller.control_calculate(bicycle.observer().state(), interp);
-            if (assistance_fade_counter < assistance_fade_period) {
-                ++assistance_fade_counter;
+            if (bicycle.v() < assistance_velocity_limit) {
```

Comment:
Actually, looks like it's fine. I'll refactor it so it's a bit easier to read though.